lib/core/versions/latest/TransactionSelector.ts

```diff
@@ -71,8 +58,33 @@ export default class TransactionSelector implements ITransactionSelector {
     return transactionsToReturn;
   }
 
-  private async getNumberOfOperationsAndTransactionsAlreadyInBlock (blockHeight: number): Promise<number[]> {
-    const transactions = await this.transactionStore.getTransactionsStartingFrom(blockHeight, blockHeight);
+  private static validateTransactions (transactions: TransactionModel[], currentTransactionTime: number) {
+    for (const transaction of transactions) {
+      // expect all transactions to be in the same transaction time
+      if (transaction.transactionTime !== currentTransactionTime) {
+        throw new SidetreeError(ErrorCode.TransactionsNotInSameBlock, 'transaction must be in the same block to perform rate limiting, investigate and fix');
+      }
+    }
+  }
+
+  // if multiple transactions have the same writer, take the first one in the array and enqueue into trasnactionPriorityQueue
```

Comment:
Nit: commenting style outside of method is with `/**`. Consider move '//' comments within the method.